scripts/build-post-list.ts

```diff
@@ -27,10 +27,14 @@ const releaseNotes: (string | undefined)[] = [];
 const HEADING_ID_REGEX = /[\s]*(?:\{#([a-zA-Z0-9\-_]+)\}|<a[\s]+name="([a-zA-Z0-9\-_]+)")/;
 
 /**
- * Slugifies a string for use in a table of contents.
+ * Extracts a slug from a markdown heading string for table of contents usage.
  *
- * @param {string} str - The string to slugify.
- * @returns {string} - The slugified string.
+ * This function searches for a valid heading ID in the form `{#someId}` within the input string
+ * and returns the trimmed ID as the slug. If the input is not a string, is empty, or does not
+ * contain a valid heading ID, an empty string is returned.
+ *
+ * @param str - The input markdown string that may contain a heading ID.
+ * @returns The extracted slug used for table of contents, or an empty string if no valid ID is found.
```

Comment:
Should add return type here as it is not explicitly mentioned in the function definition.